src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1377,38 +1377,36 @@ Namespace Microsoft.CodeAnalysis.Operations
                 Debug.Assert(boundRaiseEventStatement.HasErrors)
                 Dim children As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(Function() ImmutableArray.Create(Create(boundRaiseEventStatement.EventInvocation)))
                 Return New LazyInvalidOperation(children, _semanticModel, syntax, type, constantValue, isImplicit)
             End If
 
-            Dim receiver = eventInvocation.ReceiverOpt
-            Dim eventSymbol = boundRaiseEventStatement.EventSymbol
-            Dim eventReferenceSyntax = receiver.Syntax
+            Dim receiverOpt = eventInvocation.ReceiverOpt
+            Dim eventReferenceSyntax = If(receiverOpt?.Syntax,
+                                          If(TryCast(syntax, RaiseEventStatementSyntax)?.Name,
+                                             syntax))
             Dim eventReferenceType As ITypeSymbol = eventSymbol.Type
-            Dim eventReferenceConstantValue As [Optional](Of Object) = ConvertToOptional(receiver.ConstantValueOpt)
+            Dim eventReferenceConstantValue As [Optional](Of Object) = ConvertToOptional(receiverOpt?.ConstantValueOpt)
             ' EventReference in a raise event statement is never implicit. However, the way it is implemented, we don't get
             ' a "BoundEventAccess" for either field backed event or custom event, and the bound nodes we get are marked as
             ' generated by compiler. As a result, we have to explicitly set IsImplicit to false.
             Dim eventReferenceIsImplicit As Boolean = False
 
-            Dim boundInstance As BoundNode
-            If receiver.Kind = BoundKind.FieldAccess Then
+            If receiverOpt?.Kind = BoundKind.FieldAccess Then
                 ' For raising a field backed event, we will only get a field access node in bound tree.
-                Dim eventFieldAccess = DirectCast(receiver, BoundFieldAccess)
+                Dim eventFieldAccess = DirectCast(receiverOpt, BoundFieldAccess)
                 Debug.Assert(eventFieldAccess.FieldSymbol.AssociatedSymbol = eventSymbol)
 
-                boundInstance = eventFieldAccess.ReceiverOpt
-            Else
-                ' This is for a custom event
-                boundInstance = receiver
+                receiverOpt = eventFieldAccess.ReceiverOpt
             End If
 
-            Dim eventReferenceInstance As Lazy(Of IOperation) = CreateReceiverOperation(boundInstance, eventSymbol)
+            Dim eventReferenceInstance As Lazy(Of IOperation) = CreateReceiverOperation(receiverOpt, eventSymbol)
 
             Dim eventReference As Lazy(Of IEventReferenceOperation) = New Lazy(Of IEventReferenceOperation)(Function() As IEventReferenceOperation
                                                                                                                 Return New LazyEventReferenceExpression(eventSymbol,
```

Comment:
Is this value of the Lazy(Of IEventReferenceOperation) also a Lazy(Of IEventReferenceOperation)? If so, is the extra laziness required?

Reply:
> Is this value of the Lazy(Of IEventReferenceOperation) also a Lazy(Of IEventReferenceOperation)?

Could you clarify the question please?

Reply:
Never mind, I misread the lambda.